src/sidebar/helpers/mentions.ts

```diff
@@ -180,3 +181,27 @@ export function getContainingMentionOffsets(
         : referencePosition + subsequentCharPos,
   };
 }
+
+/**
+ * Whether mentions should be done via username (`@username`) or display name
+ * (`@[Display Name]`).
+ *
+ * This also affects the information displayed in suggestions, which will not
+ * include the username in the second case.
+ */
+export type MentionMode = 'username' | 'display-name';
+
+/**
+ * Converts provided user into a plain-text mention (not wrapped in a mention
+ * tag).
+ *
+ * Depending on the mention mode it will return `@username` or `@[Display Name]`.
+ */
+export function toPlainTextMention(
+  user: UserItem,
+  mentionMode: MentionMode,
+): string {
+  return mentionMode === 'display-name'
+    ? `@[${user.displayName ?? ''}]`
```

Comment:
There is no escaping of `[` or `]` in display names here. Do you want to address that here or in a follow up? If in a follow-up I would add a brief comment about it here.

Reply:
"Escaping" could also involve replacing or stripping these characters.

Reply:
Hmm, I didn't think a user could use square brackets in their display name, but that's definitely a possibility.

Reply:
I created an issue to investigate this https://github.com/hypothesis/client/issues/6891